Setup and run RuboCop

mokagio opened the pull request:
This PR only includes the changes made by running `bundle exec rubocop -A`, that is, RuboCop with auto-correction.

I'm happy to add further refinements based on suggestions, but I'd like to do so in a followup PR. I'll have to open one anyway because there are leftover violations that could not be automatically fixed.

This is a long PR, but it's one with _only automated changes_. As such, I think a quick review and merge is acceptable. Of course, RuboCop is not immune to bugs, but all the changes look consistent to me. We can discuss tweaks inline, but to contest with how a change is made would be to argue against RuboCop the tool, and it can't reply to us.

## Regression Notes

1. Potential unintended areas of impact – N.A.
2. What I did to test those areas of impact (or what existing automated tests I relied on) – N.A.
3. What automated tests I added (or what prevented me from doing so) – N.A.

---

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes. **N.A.**
- [x] I have considered adding accessibility improvements for my changes. **N.A.**
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. **N.A.**

mokagio committed:
Install RuboCop via Bundler

mokagio committed:
Run `Gemfile` through `bundle exec rubocop -A`

mokagio committed:
Run `Podfile` through `bundle exec rubocop -A`

mokagio committed:
Run the `fastlane` folder through `bundle exec rubocop -A`

mokagio committed:
Add `rubocop-rake` gem

mokagio committed:
Run `Rakefile` through `bundle exec rubocop -A`

mokagio committed:
Run `scripts` folder through `bundle exec rubocop -A`

mokagio commented on Rakefile:
```diff
@@ -1,29 +1,32 @@
-XCODE_CONFIGURATION='Debug'.freeze
+# frozen_string_literal: true
+
+require 'English'
```
TIL about the [English library](https://ruby-doc.org/stdlib-2.7.6/libdoc/English/rdoc/English.html)

> Include the English library file in a Ruby script, and you can reference the global variables such as VAR{$_} using less cryptic names

AliSoftware replied:
Nice, TIL too

mokagio committed:
Update `.rubocop.yml` to exclude certain folders

AliSoftware reviewed (approved):
"Hide whitespace" option saving the day for this kind of long PR diffs 😛 
<img width="264" alt="image" src="https://user-images.githubusercontent.com/216089/172649468-f42cf954-5464-4bcc-93bb-8d3c3d16a839.png">

> I'm happy to add further refinements based on suggestions, but I'd like to do so in a followup PR

Yes, totally, right choice there! 👍 

> to argue against RuboCop the tool, and it can't reply to us.

Be careful what you wish for, one day it will become sentient or get some AI and be able to start arguing 😂 

---

The most common thing that could accidentally break Ruby code when you `rubocop -A` is if you happen to mutate string literals (`x = 'hello'` then later `x += ' world'`) in your ruby script: Since `rubocop -A` tends to add `# frozen_strings_literal: true` at the top of each ruby file but does _not_ check for you if all of the strings literals used in the script are not currently accidentally mutated, that means that you could end up with a crash of your ruby script after that change. So it's something to be mindful of when we run `rubocop -A` and let it add those, making some literals which were previously mutable suddenly become frozen.

I haven't taken the time, for the files for which `rubocop -A` added that `# frozen_string_literal: true` at their top, to review in details the entire content of those files (as opposed to just the lines changed and appearing in this diff on GitHub) to see if we had some accidental instance of a mutating somewhere in those scripts. Hopefully not, and worth case scenario we can always fix these later when we discover them while running the rake tasks or fastlane lanes later in the sprint though, but the Release Manager (luckily that's you for WPiOS!) might need to be informed of that possibility though.

AliSoftware commented:
Note: apparently most of the CI failures above are because `` `Podfile.lock` was changed during `pod install` ``. Which I'm guessing is just because the checksum of the `Podfile` was changed (because of `rubocop -A` modifying the file's content). So I'm assuming re-running `pod install` and commit the resulting change (which should just be on the `CHECKSUM` entry of the `Podfile.lock`) would suffice. But I'll let you do that because I see your PR is still in Draft so I'm not sure if you planned to add more commits and make more changes to the `Podfile` (and thus would need to re-run `pod install` once again) before undrafting it.

mokagio commented:
> Be careful what you wish for, one day it will become sentient or get some AI and be able to start arguing 😂

Golly gosh! With a face like this, I definitely would argue with it:

<img width="707" alt="image" src="https://user-images.githubusercontent.com/1218433/172744575-b9b7eee4-92c4-4fa0-841a-a872a109e607.png">

mokagio committed:
Track result of running `bundle exec pod install`

I should have done this as part of
ec84fc6b21a256e6b7c3b776db43a3cf68da7050, but I forgot.

Thank you CI for picking this up 🙇‍♂️ And @AliSoftware, too, for
mentioning it.

See
https://github.com/wordpress-mobile/WordPress-iOS/pull/18839#issuecomment-1150056516

mokagio commented:
> [CI fails] because `Podfile.lock` was changed during `pod install`.

Addressed in [31a54813877dd129c5e90ca8efd76b8ac5ec6d4a](https://github.com/wordpress-mobile/WordPress-iOS/pull/18839/commits/31a54813877dd129c5e90ca8efd76b8ac5ec6d4a). Enabling auto-merge to move forward.

mokagio commented:
For completeness and to solidify the lesson learned: Yep, the migration broke something, see https://github.com/wordpress-mobile/WordPress-iOS/pull/18850.

Joke's on me for not running a few `rake` tasks after passing `Rakefile` through RuboCop. _Trust... but verify!_
